framework/src/transfers/MultiAppGeneralFieldNearestLocationTransfer.C

```diff
@@ -436,3 +513,108 @@ MultiAppGeneralFieldNearestLocationTransfer::inBlocks(const std::set<SubdomainID
   }
   return false;
 }
+
+unsigned int
+MultiAppGeneralFieldNearestLocationTransfer::getNumSources() const
+{
+  // Number of source = number of KDTrees.
+  // Using mesh divisions or nearest-positions, for every app we use 1 tree per division
+  if (!_from_mesh_divisions.empty() || (_nearest_positions_obj && !_group_subapps))
+    return _from_problems.size() * getNumDivisions();
+  // If we group apps, then we only use one tree per division (nearest-position region)
+  else if (_nearest_positions_obj && _group_subapps)
+    return _nearest_positions_obj->getNumPositions(_fe_problem.getCurrentExecuteOnFlag() ==
+                                                   EXEC_INITIAL);
+  // Regular case: 1 KDTree per app
+  else
+    return _from_problems.size();
+}
+
+Point
+MultiAppGeneralFieldNearestLocationTransfer::getPointInLocalSourceFrame(unsigned int i_from,
+                                                                        const Point & pt) const
+{
+
+  if (!_nearest_positions_obj &&
+      (!_from_transforms[getGlobalSourceAppIndex(i_from)]->hasCoordinateSystemTypeChange() ||
+       _skip_coordinate_collapsing))
+    return _from_transforms[getGlobalSourceAppIndex(i_from)]->mapBack(pt);
```

Comment:
So if there is a coordinate system type change then the transform will never be used? Is there an option for the user to change that behavior if they wanted to? Also it seems that if `_skip_coordinate_collapsing = true`, then the transform object will never be used even if there are no coordinate system type changes. It's possible I'm reading the condition wrong

Reply:
if _skip_coordinate_collapsing == true, then the first if-condition is true and we do:
```
    return _from_transforms[getGlobalSourceAppIndex(i_from)]->mapBack(pt);
```
so it should be used. The from transformed will have been set to skip the coordinate collapsing that will skip still.

mapBack errors on going from RZ to XYZ, so this precludes using it. I agree beta/alpha/gamma rotation + scaling is being ignored in that case. We could add more APIs to coordinate transforms to probe more what's in the transform and trigger only the parts we want

Reply:
if we had an API like "do the transform map back except if it will error then dont do that erroring part but do everything else" I would use that.

Reply:
👍